# Etude de plusieurs facteurs environnementaux sur la production dans les hôpitaux

In [1]:
# importation des packages
import os 
import pandas as pd
import numpy as np
import seaborn as sns
import matplotlib.pyplot as plt
import statsmodels.api as sm

C:\Users\Vinciane\Anaconda3\lib\site-packages\statsmodels\tools\_testing.py:19: FutureWarning: pandas.util.testing is deprecated. Use the functions in the public API at pandas.testing instead.
  import pandas.util.testing as tm


In [2]:
VAR_CTRL = pd.read_csv('https://raw.githubusercontent.com/VincianeDesbois/Hopitaux_Production/main/Variables_controle.csv',  sep = ',')
VAR_EXPL_CAT  = pd.read_csv('https://raw.githubusercontent.com/VincianeDesbois/Hopitaux_Production/main/VAR_EXPL_CAT.csv',  sep = ',')

In [7]:
VAR_EXPL_CAT

Unnamed: 0 Numero_finess  Médecins libéraux  \
0              1     010008407                0.0   
1              2     010009132                0.0   
2              3     010780054                4.5   
3              4     010780062                0.0   
4              5     010780096                0.0   
...          ...           ...                ...   
1147        1117     940000649                0.0   
1148        1118     940000656                0.0   
1149        1119     940000664                0.0   
1150        1135     950000406                0.0   
1151        1150     950630012                0.0   

      Autres personnels médicaux libéraux  Médecins salariés  \
0                                     0.0             18.490   
1                                     0.0              1.795   
2                                     1.0            102.600   
3                                     0.0             19.450   
4                                     0.0             10.260   
...                                   ...                ...   
1147                                  0.0             97.680   
1148                                  0.0              9.460   
1149                                  0.0            202.110   
1150                                  0.0              3.510   
1151                                  0.0             11.300   

      Autres personnels médicaux salariés  \
0                                  14.330   
1                                   0.070   
2                                  44.530   
3                                  10.100   
4                                   1.885   
...                                   ...   
1147                               28.690   
1148                                1.000   
1149                               79.550   
1150                                1.260   
1151                                2.500   

      Personnels de direction et administratifs  \
0                                     21.196667   
1                                      4.175000   
2                                    118.605000   
3                                     58.450000   
4                                      9.685000   
...                                         ...   
1147                                 132.300000   
1148                                  39.470000   
1149                                 459.580000   
1150                                  13.950000   
1151                                  23.070000   

      Personnels des services de soins  Personnels éducatifs et sociaux  \
0                            97.323333                         0.846667   
1                            46.700000                         1.065000   
2                           388.590000                         4.930000   
3                           336.540000                         5.450000   
4                            88.475000                         2.555000   
...                                ...                              ...   
1147                        468.280000                         4.830000   
1148                         93.680000                         2.520000   
1149                        904.180000                         9.700000   
1150                         78.740000                         1.980000   
1151                        177.880000                        39.070000   

      Personnels médico-techniques  Personnels techniques et ouvriers  \
0                         9.206667                             26.440   
1                         1.130000                             13.465   
2                       107.660000                            112.630   
3                        29.320000                             57.570   
4                         2.570000                              5.960   
...                            ...                                ...   
1147                     58.750000

# Première analyse sur l'effet du revenu fiscal moyen sur les points ISA

In [166]:
VAR_EXPL_CAT_1 = VAR_EXPL_CAT[['Numero_finess', 'Personnel médical','STJR_1','STJR_2', 'STJR_3',
       'Personnels de direction et administratifs',
       'Personnels des services de soins', 'Personnels éducatifs et sociaux',
       'Personnels médico-techniques', 'Personnels techniques et ouvriers',
        'Internes']]
VAR_EXPL_CAT_1= VAR_EXPL_CAT_1.fillna(0)

In [167]:
VAR_EXPL_CAT_1

Numero_finess  Personnel médical  STJR_1  STJR_2  STJR_3  \
0        010008407             32.820     1.0     0.0     0.0   
1        010009132              1.865     1.0     0.0     0.0   
2        010780054            152.630     1.0     0.0     0.0   
3        010780062             29.550     1.0     0.0     0.0   
4        010780096             12.145     1.0     0.0     0.0   
...            ...                ...     ...     ...     ...   
1147     940000649            126.370     0.0     0.0     1.0   
1148     940000656             10.460     0.0     0.0     1.0   
1149     940000664            281.660     0.0     0.0     1.0   
1150     950000406              4.770     0.0     0.0     1.0   
1151     950630012             13.800     0.0     0.0     1.0   

      Personnels de direction et administratifs  \
0                                     21.196667   
1                                      4.175000   
2                                    118.605000   
3                                     58.450000   
4                                      9.685000   
...                                         ...   
1147                                 132.300000   
1148                                  39.470000   
1149                                 459.580000   
1150                                  13.950000   
1151                                  23.070000   

      Personnels des services de soins  Personnels éducatifs et sociaux  \
0                            97.323333                         0.846667   
1                            46.700000                         1.065000   
2                           388.590000                         4.930000   
3                           336.540000                         5.450000   
4                            88.475000                         2.555000   
...                                ...                              ...   
1147                        468.280000                         4.830000   
1148                         93.680000                         2.520000   
1149                        904.180000                         9.700000   
1150                         78.740000                         1.980000   
1151                        177.880000                        39.070000   

      Personnels médico-techniques  Personnels techniques et ouvriers  \
0                         9.206667                             26.440   
1                         1.130000                             13.465   
2                       107.660000                            112.630   
3                        29.320000                             57.570   
4                         2.570000                              5.960   
...                            ...                                ...   
1147                     58.750000                             25.760   
1148                      1.060000                             51.490   
1149                    404.530000                            126.770   
1150                      5.580000                              2.360   
1151                      5.330000                             12.700   

      Internes  
0         0.00  
1         0.00  
2       195.09  
3         6.45  
4         3.48  
...        ...  
1147    132.00  
1148      1.47  
1149    402.00  
1150      0.00  
1151      2.49  

[1152 rows x 11 columns]

In [168]:
VAR_EXPL_CAT_1 = VAR_EXPL_CAT_1.apply(lambda x: np.log(x) if x.name in ['Personnel médical',
       'Personnels de direction et administratifs',
       'Personnels des services de soins', 'Personnels éducatifs et sociaux',
       'Personnels médico-techniques', 'Personnels techniques et ouvriers',
        'Internes'] else x)

from numpy import inf
VAR_EXPL_CAT_1.replace(-inf, inf, inplace = True)

for i in ['Personnels de direction et administratifs','Personnels des services de soins', 'Personnels éducatifs et sociaux',
                                                    'Personnels médico-techniques', 'Personnels techniques et ouvriers',
                                                    'Personnel médical', 'Internes']:
    VAR_EXPL_CAT_1[i]=VAR_EXPL_CAT_1[i].replace(inf, VAR_EXPL_CAT_1[i].min() - 0.01)


C:\Users\Vinciane\Anaconda3\lib\site-packages\pandas\core\arraylike.py:358: RuntimeWarning: divide by zero encountered in log
  result = getattr(ufunc, method)(*inputs, **kwargs)


In [169]:
VAR_EXPL_CAT_1

Numero_finess  Personnel médical  STJR_1  STJR_2  STJR_3  \
0        010008407           3.491038     1.0     0.0     0.0   
1        010009132           0.623261     1.0     0.0     0.0   
2        010780054           5.028017     1.0     0.0     0.0   
3        010780062           3.386084     1.0     0.0     0.0   
4        010780096           2.496918     1.0     0.0     0.0   
...            ...                ...     ...     ...     ...   
1147     940000649           4.839214     0.0     0.0     1.0   
1148     940000656           2.347558     0.0     0.0     1.0   
1149     940000664           5.640701     0.0     0.0     1.0   
1150     950000406           1.562346     0.0     0.0     1.0   
1151     950630012           2.624669     0.0     0.0     1.0   

      Personnels de direction et administratifs  \
0                                      3.053844   
1                                      1.429114   
2                                      4.775799   
3                                      4.068172   
4                                      2.270578   
...                                         ...   
1147                                   4.885072   
1148                                   3.675541   
1149                                   6.130313   
1150                                   2.635480   
1151                                   3.138533   

      Personnels des services de soins  Personnels éducatifs et sociaux  \
0                             4.578039                        -0.166448   
1                             3.843744                         0.062975   
2                             5.962525                         1.595339   
3                             5.818717                         1.695616   
4                             4.482720                         0.938052   
...                                ...                              ...   
1147                          6.149066                         1.574846   
1148                          4.539885                         0.924259   
1149                          6.807028                         2.272126   
1150                          4.366151                         0.683097   
1151                          5.181109                         3.665355   

      Personnels médico-techniques  Personnels techniques et ouvriers  \
0                         2.219928                           3.274878   
1                         0.122218                           2.600094   
2                         4.678978                           4.724108   
3                         3.378270                           4.053002   
4                         0.943906                           1.785070   
...                            ...                                ...   
1147                      4.073291                           3.248823   
1148                      0.058269                           3.941388   
1149                      6.002726                           4.842374   
1150                      1.719189                           0.858662   
1151                      1.673351                           2.541602   

      Internes  
0    -1.437116  
1    -1.437116  
2     5.273461  
3     1.864080  
4     1.247032  
...        ...  
1147  4.882802  
1148  0.385262  
1149  5.996452  
1150 -1.437116  
1151  0.912283  

[1152 rows x 11 columns]

In [170]:
VAR_CTRL.columns
#Je prends Cpaital_LIT_MCO et Capital_Bloc

Index(['Numero_finess', 'Points_ISA', 'stjr', 'dep', 'Departement', 'COMINSEE',
       'Capital_Lit_MCO', 'Capital_Bloc', 'Capital_imagerie',
       'Capital_Examens_biologiques', 'Capital_telemed', 'SEJHC_MED',
       'SEJHC_CHI', 'SEJHC_OBS', 'SEJHC_MCO', 'Taux_chomage_par_dep',
       'Population', 'Dep Moyenne Salaires Horaires',
       'Dep Moyenne Salaires Cadre Horaires',
       'Dep Moyenne Salaires Prof Intermédiaire Horaires',
       'Dep Moyenne Salaires Employé Horaires',
       'Moyenne Revenus Fiscaux Départementaux', 'Bloc_politique'],
      dtype='object')

In [171]:
VAR_CTRL_1 = VAR_CTRL.copy()

VAR_CTRL_1 = VAR_CTRL_1.apply(lambda x: np.log(x) if x.name in ['Points_ISA', 'Capital_Lit_MCO', 'Capital_Bloc', 'Capital_imagerie',
       'Capital_Examens_biologiques', 'Capital_telemed', 'Population'] else x)


from numpy import inf
VAR_CTRL_1.replace(-inf, inf, inplace = True)

for i in ['Points_ISA', 'Capital_Lit_MCO', 'Capital_Bloc', 'Capital_imagerie',
       'Capital_Examens_biologiques', 'Capital_telemed', 'Population']:
    VAR_CTRL_1[i]=VAR_CTRL_1[i].replace(inf, VAR_CTRL_1[i].min() - 0.01)


C:\Users\Vinciane\Anaconda3\lib\site-packages\pandas\core\arraylike.py:358: RuntimeWarning: divide by zero encountered in log
  result = getattr(ufunc, method)(*inputs, **kwargs)


In [172]:
VAR_CTRL.describe()

Points_ISA         stjr  Capital_Lit_MCO  Capital_Bloc  \
count  1.162000e+03  1162.000000      1139.000000   1153.000000   
mean   2.939915e+05     1.707401       125.109263      7.379577   
std    6.281782e+05     0.716795       127.684194      7.191520   
min    1.347900e+02     1.000000         0.000000      0.000000   
25%    2.993899e+04     1.000000        35.000000      1.000000   
50%    1.016205e+05     2.000000        85.000000      6.000000   
75%    2.975944e+05     2.000000       171.166667     10.500000   
max    9.116903e+06     3.000000      1055.000000     44.000000   

       Capital_imagerie  Capital_Examens_biologiques  Capital_telemed  \
count       1162.000000                 1.149000e+03      1162.000000   
mean           4.256971                 2.508881e+05         0.728148   
std            5.047103                 6.567670e+05         1.514649   
min            0.000000                 0.000000e+00         0.000000   
25%            0.000000                 0.000000e+00         0.000000   
50%            3.000000                 0.000000e+00         0.000000   
75%            7.000000                 6.261400e+04         1.000000   
max           28.000000                 1.060114e+07        11.000000   

          SEJHC_MED     SEJHC_CHI    SEJHC_OBS     SEJHC_MCO  \
count   1162.000000   1162.000000  1161.000000   1162.000000   
mean    3953.236728   2091.528256   673.093714   6717.279443   
std     5172.505167   2471.292213  1146.876402   7600.700172   
min        0.000000      0.000000     0.000000      0.000000   
25%      364.750000      0.000000     0.000000   1204.500000   
50%     1963.000000   1428.000000     0.000000   4168.500000   
75%     5581.250000   3168.312500  1081.000000   9749.750000   
max    39884.000000  20991.000000  7639.000000  60719.000000   

       Taux_chomage_par_dep    Population  Dep Moyenne Salaires Horaires  \
count           1162.000000  1.162000e+03                    1162.000000   
mean               8.290103  1.367009e+05                      11.793749   
std                1.660836  3.998901e+05                       1.908702   
min                5.000000  1.820000e+02                       9.862500   
25%                7.000000  8.053750e+03                      10.633333   
50%                8.000000  2.295500e+04                      11.204762   
75%                9.300000  6.171825e+04                      11.981429   
max               13.800000  2.125851e+06                      17.800000   

       Dep Moyenne Salaires Cadre Horaires  \
count                          1162.000000   
mean                             21.764363   
std                               1.674054   
min                              19.750000   
25%                              20.775000   
50%                              21.280808   
75%                              22.083750   
max                              27.600000   

       Dep Moyenne Salaires Prof Intermédiaire Horaires  \
count                                       1162.000000   
mean                                          12.674821   
std                                            0.523229   
min                                           11.740000   
25%                                           12.346721   
50%                                           12.535714   
75%                                           12.902500   
max                                           14.185000   

       Dep Moyenne Salaires Employé Horaires  \
count                            1162.000000   
mean                                8.783985   
std                                 0.353571   
min                                 8.261905   
25%                                 8.555172   
50%                                 8.700862   
75%                                 8.848113   
max                                 9.742553   

       Moyenne Revenus Fiscaux Départementaux  
count                             1162.000000  
mean     

Le premier quartile des revenus fiscaux moyens par mois est à 9471, et le troisième quartile à 11536. On sépare en deux bases : celles avec les départements avec les plus faibles revenus, et celles avec les départements avec les plus hauts revenus.

In [173]:
# Les établissements avec des revenus départements  trés élevés

VAR_CTRL_ELEVES = VAR_CTRL_1.copy()
VAR_CTRL_ELEVES= VAR_CTRL_ELEVES.rename(columns = {'Moyenne Revenus Fiscaux Départementaux': 'Rev_dep'})

VAR_CTRL_ELEVES= VAR_CTRL_ELEVES[VAR_CTRL_ELEVES.Rev_dep >= 10239]

In [174]:
# Les établissements avec des revenus départements trésfaibles

VAR_CTRL_FAIBLES = VAR_CTRL_1.copy()
VAR_CTRL_FAIBLES = VAR_CTRL_FAIBLES.rename(columns = {'Moyenne Revenus Fiscaux Départementaux': 'Rev_dep'})

VAR_CTRL_FAIBLES = VAR_CTRL_FAIBLES[VAR_CTRL_FAIBLES.Rev_dep <= 10239]

In [175]:
VAR_CTRL_FAIBLES

Numero_finess  Points_ISA  stjr dep       Departement COMINSEE  \
20       130001647   13.658609     3  13  Bouches-du-Rhône    13055   
21       130784051   12.784332     2  13  Bouches-du-Rhône    13055   
22       130783152    8.674199     3  13  Bouches-du-Rhône    13055   
23       130783665   11.152187     3  13  Bouches-du-Rhône    13055   
24       130785652   14.297202     3  13  Bouches-du-Rhône    13055   
...            ...         ...   ...  ..               ...      ...   
1157     120780085   10.729103     1  12           Aveyron    12089   
1158     480780097   12.391985     1  48            Lozère    48095   
1159     480780162    8.648072     1  48            Lozère    48080   
1160     480780121    7.781744     1  48            Lozère    48140   
1161     480780154    6.249357     1  48            Lozère    48092   

      Capital_Lit_MCO  Capital_Bloc  Capital_imagerie  \
20           5.602119      2.484907          2.564949   
21           5.690359      2.944439          2.564949   
22           2.484907     -0.010000         -0.010000   
23           3.891820      1.609438          1.791759   
24           6.519147      3.367296          2.639057   
...               ...           ...               ...   
1157         3.526361      1.609438          1.098612   
1158         4.204693      1.504077          1.791759   
1159         3.135494     -0.010000          0.000000   
1160         3.091042     -0.010000         -0.010000   
1161         2.995732     -0.010000         -0.010000   

      Capital_Examens_biologiques  ...  SEJHC_OBS  SEJHC_MCO  \
20                       9.733292  ...        0.0    15228.0   
21                       5.707028  ...        0.0    12231.0   
22                       5.707028  ...        0.0      419.0   
23                      11.171041  ...        0.0     1525.0   
24                      12.740875  ...     6414.0    40357.0   
...                           ...  ...        ...        ...   
1157                    12.367630  ...        0.0     2056.0   
1158                    12.956158  ...      392.0     3599.5   
1159                     5.707028  ...        0.0      421.0   
1160                     5.707028  ...        0.0      205.0   
1161                     5.707028  ...        0.0      384.0   

      Taux_chomage_par_dep  Population  Dep Moyenne Salaires Horaires  \
20                    10.1   13.589226                      12.477895   
21                    10.1   13.589226                      12.477895   
22                    10.1   13.589226                      12.477895   
23                    10.1   13.589226                      12.477895   
24                    10.1   13.589226                      12.477895   
...                    ...         ...                            ...   
1157                   6.4    8.822764                      10.115789   
1158                   5.3    9.375177                      10.075000   
1159                   5.3    8.038189                      10.075000   
1160                   5.3    8.370779                      10.075000   
1161                   5.3    8.613230                      10.075000   

      Dep Moyenne Salaires Cadre Horaires  \
20                              22.122105   
21                              22.122105   
22                              22.122105   
23                              22.122105   
24                              22.122105   
...                                   ...   
1157                            20.252632   
1158                            20.475000   
1159                            20.475000   
1160                            20.475000   
1161                            20.475000   

      Dep Moyenne Salaires Prof Intermédiaire Horaires  \
20                                           13.044211   
21                                           13.044211   
22                                           13.044211   
23                                           13.

In [176]:
VAR_CTRL_FAIBLES= VAR_CTRL_FAIBLES.fillna(0)
VAR_CTRL_ELEVES= VAR_CTRL_ELEVES.fillna(0)
VAR_EXPL_CAT_1= VAR_EXPL_CAT_1.fillna(0)

In [177]:
analyse_dep_faibles = pd.merge(VAR_CTRL_FAIBLES, VAR_EXPL_CAT_1)

analyse_dep_eleves = pd.merge(VAR_CTRL_ELEVES, VAR_EXPL_CAT_1)

In [178]:
analyse_dep_faibles.columns

Index(['Numero_finess', 'Points_ISA', 'stjr', 'dep', 'Departement', 'COMINSEE',
       'Capital_Lit_MCO', 'Capital_Bloc', 'Capital_imagerie',
       'Capital_Examens_biologiques', 'Capital_telemed', 'SEJHC_MED',
       'SEJHC_CHI', 'SEJHC_OBS', 'SEJHC_MCO', 'Taux_chomage_par_dep',
       'Population', 'Dep Moyenne Salaires Horaires',
       'Dep Moyenne Salaires Cadre Horaires',
       'Dep Moyenne Salaires Prof Intermédiaire Horaires',
       'Dep Moyenne Salaires Employé Horaires', 'Rev_dep', 'Bloc_politique',
       'Personnel médical', 'STJR_1', 'STJR_2', 'STJR_3',
       'Personnels de direction et administratifs',
       'Personnels des services de soins', 'Personnels éducatifs et sociaux',
       'Personnels médico-techniques', 'Personnels techniques et ouvriers',
       'Internes'],
      dtype='object')

In [179]:
analyse_dep_faibles_X =analyse_dep_faibles[["Personnel médical","Internes",'Personnels de direction et administratifs',
       'Personnels des services de soins', 'Personnels éducatifs et sociaux',
       'Personnels médico-techniques', 'Personnels techniques et ouvriers', 
        'Capital_Lit_MCO', 'Capital_Bloc', 'Capital_imagerie', 'Population',
       'Capital_Examens_biologiques', 'Capital_telemed', 'STJR_1', 'STJR_2', 'STJR_3']]
analyse_dep_faibles_Y =analyse_dep_faibles[['Points_ISA']]
analyse_dep_eleves_X =analyse_dep_eleves[["Personnel médical","Internes",'Personnels de direction et administratifs',
       'Personnels des services de soins', 'Personnels éducatifs et sociaux',
       'Personnels médico-techniques', 'Personnels techniques et ouvriers', 
       'Capital_Lit_MCO', 'Capital_Bloc', 'Capital_imagerie', 'Population',
       'Capital_Examens_biologiques', 'Capital_telemed', 'STJR_1', 'STJR_2', 'STJR_3']]
analyse_dep_eleves_Y =analyse_dep_eleves[['Points_ISA']]

In [180]:
analyse_dep_faibles_X 

Personnel médical  Internes  Personnels de direction et administratifs  \
0             5.249337  5.499502                                   5.538593   
1             4.474036 -1.437116                                   3.810655   
2             1.818077 -1.437116                                   2.631889   
3             4.042876 -1.437116                                   3.246491   
4             5.447599  3.988984                                   5.749966   
..                 ...       ...                                        ...   
575           3.231989  0.667829                                   3.411148   
576           4.069283  2.890372                                   3.539993   
577           1.360977 -1.437116                                   1.945910   
578           1.909543 -1.437116                                   0.565314   
579           1.388791 -1.437116                                   1.249902   

     Personnels des services de soins  Personnels éducatifs et sociaux  \
0                            6.436006                         2.770086   
1                            6.137317                         0.000000   
2                            4.009331                         0.000000   
3                            4.380400                         0.000000   
4                            7.219056                         1.908060   
..                                ...                              ...   
575                          5.165072                         0.693147   
576                          4.997888                         1.483875   
577                          3.837299                         0.000000   
578                          3.149311                        -3.922023   
579                          3.651697                        -1.714798   

     Personnels médico-techniques  Personnels techniques et ouvriers  \
0                        5.615352                           4.537641   
1                        2.771338                           3.171365   
2                       -1.609438                           0.916291   
3                        2.240710                           2.484907   
4                        4.879691                           4.835170   
..                            ...                                ...   
575                      3.397858                           3.632309   
576                      2.924236                           3.096256   
577                      0.000000                           2.420368   
578                     -1.386294                           1.324419   
579                     -3.228876                           1.916923   

     Capital_Lit_MCO  Capital_Bloc  Capital_imagerie  Population  \
0           5.602119      2.484907          2.564949   13.589226   
1           5.690359      2.944439          2.564949   13.589226   
2           2.484907     -0.010000         -0.010000   13.589226   
3           3.891820      1.609438          1.791759   13.589226   
4           6.519147      3.367296          2.639057   13.589226   
..               ...           ...               ...         ...   
575         3.526361      1.609438          1.098612    8.822764   
576         4.204693      1.504077          1.791759    9.375177   
577         3.135494     -0.010000          0.000000    8.038189   
578         3.091042     -0.010000         -0.010000    8.370779   
579         2.995732     -0.010000         -0.010000    8.613230   

     Capital_Examens_biologiques  Capital_telemed  STJR_1  STJR_2  STJR_3  
0                       9.733292        -0.703147     0.0     0.0     1.0  
1                       5.707028        -0.703147     0.0     1.0     0.0  
2                       5.707028         0.000000     0.0     0.0     1.0  
3                      11.171041        -0.703147     0.0     0.0     1.0  
4                      12.740875        -0.703147     0.0     0.0     1.0  
..                           

In [181]:
analyse_dep_faibles_X = sm.add_constant(analyse_dep_faibles_X) # on ajoute une constante, sinon la régression se fait sans, par défaut
y = analyse_dep_faibles[['Points_ISA']]

model_dep_faibles = sm.OLS(y, analyse_dep_faibles_X)
results_dep_faibles = model_dep_faibles.fit()
print(results_dep_faibles.summary())

                            OLS Regression Results                            
Dep. Variable:             Points_ISA   R-squared:                       0.907
Model:                            OLS   Adj. R-squared:                  0.904
Method:                 Least Squares   F-statistic:                     365.4
Date:                Sun, 16 May 2021   Prob (F-statistic):          8.31e-279
Time:                        17:08:07   Log-Likelihood:                -461.87
No. Observations:                 580   AIC:                             955.7
Df Residuals:                     564   BIC:                             1026.
Df Model:                          15                                         
Covariance Type:            nonrobust                                         
                                                coef    std err          t      P>|t|      [0.025      0.975]
----------------------------------------------------------------------------------------------------

In [182]:
analyse_dep_eleves_X = sm.add_constant(analyse_dep_eleves_X) # on ajoute une constante, sinon la régression se fait sans, par défaut
y = analyse_dep_eleves[['Points_ISA']]

model_dep_eleves = sm.OLS(y, analyse_dep_eleves_X)
results_dep_eleves = model_dep_eleves.fit()
print(results_dep_eleves.summary())

                            OLS Regression Results                            
Dep. Variable:             Points_ISA   R-squared:                       0.885
Model:                            OLS   Adj. R-squared:                  0.882
Method:                 Least Squares   F-statistic:                     296.0
Date:                Sun, 16 May 2021   Prob (F-statistic):          9.76e-259
Time:                        17:08:08   Log-Likelihood:                -522.67
No. Observations:                 591   AIC:                             1077.
Df Residuals:                     575   BIC:                             1147.
Df Model:                          15                                         
Covariance Type:            nonrobust                                         
                                                coef    std err          t      P>|t|      [0.025      0.975]
----------------------------------------------------------------------------------------------------

# Deuxième analyse : effet de la moyenne des salaires des cadres sur l'emploi de personnel médical libéral

In [183]:
VAR_EXPL_CAT_2 = VAR_EXPL_CAT.copy()

In [184]:
VAR_EXPL_CAT_2['Personnel médical salarié'] = VAR_EXPL_CAT['Médecins salariés'] + VAR_EXPL_CAT["Autres personnels médicaux salariés"]
VAR_EXPL_CAT_2['Personnel médical libéral'] = VAR_EXPL_CAT['Médecins libéraux'] + VAR_EXPL_CAT["Autres personnels médicaux libéraux"]

In [185]:
VAR_EXPL_CAT_2 = VAR_EXPL_CAT_2[['Numero_finess', 'Personnel médical salarié','Personnel médical libéral','STJR_1', 'STJR_2', 'STJR_3',
       'Personnels de direction et administratifs',
       'Personnels des services de soins', 'Personnels éducatifs et sociaux',
       'Personnels médico-techniques', 'Personnels techniques et ouvriers',
        'Internes']]
VAR_EXPL_CAT_2= VAR_EXPL_CAT_2.fillna(0)

In [186]:
VAR_EXPL_CAT_2

Numero_finess  Personnel médical salarié  Personnel médical libéral  \
0        010008407                     32.820                        0.0   
1        010009132                      1.865                        0.0   
2        010780054                    147.130                        5.5   
3        010780062                     29.550                        0.0   
4        010780096                     12.145                        0.0   
...            ...                        ...                        ...   
1147     940000649                    126.370                        0.0   
1148     940000656                     10.460                        0.0   
1149     940000664                    281.660                        0.0   
1150     950000406                      4.770                        0.0   
1151     950630012                     13.800                        0.0   

      STJR_1  STJR_2  STJR_3  Personnels de direction et administratifs  \
0        1.0     0.0     0.0                                  21.196667   
1        1.0     0.0     0.0                                   4.175000   
2        1.0     0.0     0.0                                 118.605000   
3        1.0     0.0     0.0                                  58.450000   
4        1.0     0.0     0.0                                   9.685000   
...      ...     ...     ...                                        ...   
1147     0.0     0.0     1.0                                 132.300000   
1148     0.0     0.0     1.0                                  39.470000   
1149     0.0     0.0     1.0                                 459.580000   
1150     0.0     0.0     1.0                                  13.950000   
1151     0.0     0.0     1.0                                  23.070000   

      Personnels des services de soins  Personnels éducatifs et sociaux  \
0                            97.323333                         0.846667   
1                            46.700000                         1.065000   
2                           388.590000                         4.930000   
3                           336.540000                         5.450000   
4                            88.475000                         2.555000   
...                                ...                              ...   
1147                        468.280000                         4.830000   
1148                         93.680000                         2.520000   
1149                        904.180000                         9.700000   
1150                         78.740000                         1.980000   
1151                        177.880000                        39.070000   

      Personnels médico-techniques  Personnels techniques et ouvriers  \
0                         9.206667                             26.440   
1                         1.130000                             13.465   
2                       107.660000                            112.630   
3                        29.320000                             57.570   
4                         2.570000                              5.960   
...                            ...                                ...   
1147                     58.750000                             25.760   
1148                      1.060000                             51.490   
1149                    404.530000                            126.770   
1150                      5.580000                              2.360   
1151                      5.330000                             12.700   

      Internes  
0         0.00  
1         0.00  
2       195.09  
3         6.45  
4         3.48  
...        ...  
1147    132.00  
1148      1.47  
1149    402.00  
1150      0.00  
1151      2.49  

[1152 rows x 12 columns]

In [187]:
VAR_EXPL_CAT_2 = VAR_EXPL_CAT_2.apply(lambda x: np.log(x) if x.name in ['Personnel médical salarié','Personnel médical libéral',
       'Personnels de direction et administratifs',
       'Personnels des services de soins', 'Personnels éducatifs et sociaux',
       'Personnels médico-techniques', 'Personnels techniques et ouvriers',
        'Internes'] else x)

#On remplace les valeurs -infini apparues après l'application du log par la valeur -3 c'est-à-dire log(0.001)
from numpy import inf
VAR_EXPL_CAT_2.replace(-inf, inf, inplace = True)

for i in ['Personnels de direction et administratifs','Personnels des services de soins', 'Personnels éducatifs et sociaux',
                                                    'Personnels médico-techniques', 'Personnels techniques et ouvriers',
                                                    'Personnel médical salarié','Personnel médical libéral', 'Internes']:
    VAR_EXPL_CAT_2[i]=VAR_EXPL_CAT_2[i].replace(inf, VAR_EXPL_CAT_2[i].min() - 0.01)


C:\Users\Vinciane\Anaconda3\lib\site-packages\pandas\core\arraylike.py:358: RuntimeWarning: divide by zero encountered in log
  result = getattr(ufunc, method)(*inputs, **kwargs)


In [188]:
VAR_CTRL.describe()

Points_ISA         stjr  Capital_Lit_MCO  Capital_Bloc  \
count  1.162000e+03  1162.000000      1139.000000   1153.000000   
mean   2.939915e+05     1.707401       125.109263      7.379577   
std    6.281782e+05     0.716795       127.684194      7.191520   
min    1.347900e+02     1.000000         0.000000      0.000000   
25%    2.993899e+04     1.000000        35.000000      1.000000   
50%    1.016205e+05     2.000000        85.000000      6.000000   
75%    2.975944e+05     2.000000       171.166667     10.500000   
max    9.116903e+06     3.000000      1055.000000     44.000000   

       Capital_imagerie  Capital_Examens_biologiques  Capital_telemed  \
count       1162.000000                 1.149000e+03      1162.000000   
mean           4.256971                 2.508881e+05         0.728148   
std            5.047103                 6.567670e+05         1.514649   
min            0.000000                 0.000000e+00         0.000000   
25%            0.000000                 0.000000e+00         0.000000   
50%            3.000000                 0.000000e+00         0.000000   
75%            7.000000                 6.261400e+04         1.000000   
max           28.000000                 1.060114e+07        11.000000   

          SEJHC_MED     SEJHC_CHI    SEJHC_OBS     SEJHC_MCO  \
count   1162.000000   1162.000000  1161.000000   1162.000000   
mean    3953.236728   2091.528256   673.093714   6717.279443   
std     5172.505167   2471.292213  1146.876402   7600.700172   
min        0.000000      0.000000     0.000000      0.000000   
25%      364.750000      0.000000     0.000000   1204.500000   
50%     1963.000000   1428.000000     0.000000   4168.500000   
75%     5581.250000   3168.312500  1081.000000   9749.750000   
max    39884.000000  20991.000000  7639.000000  60719.000000   

       Taux_chomage_par_dep    Population  Dep Moyenne Salaires Horaires  \
count           1162.000000  1.162000e+03                    1162.000000   
mean               8.290103  1.367009e+05                      11.793749   
std                1.660836  3.998901e+05                       1.908702   
min                5.000000  1.820000e+02                       9.862500   
25%                7.000000  8.053750e+03                      10.633333   
50%                8.000000  2.295500e+04                      11.204762   
75%                9.300000  6.171825e+04                      11.981429   
max               13.800000  2.125851e+06                      17.800000   

       Dep Moyenne Salaires Cadre Horaires  \
count                          1162.000000   
mean                             21.764363   
std                               1.674054   
min                              19.750000   
25%                              20.775000   
50%                              21.280808   
75%                              22.083750   
max                              27.600000   

       Dep Moyenne Salaires Prof Intermédiaire Horaires  \
count                                       1162.000000   
mean                                          12.674821   
std                                            0.523229   
min                                           11.740000   
25%                                           12.346721   
50%                                           12.535714   
75%                                           12.902500   
max                                           14.185000   

       Dep Moyenne Salaires Employé Horaires  \
count                            1162.000000   
mean                                8.783985   
std                                 0.353571   
min                                 8.261905   
25%                                 8.555172   
50%                                 8.700862   
75%                                 8.848113   
max                                 9.742553   

       Moyenne Revenus Fiscaux Départementaux  
count                             1162.000000  
mean     

On garde les 25% d'hopitaux avec les salaires horaires des cadres les plus bas (<20.7750, et les 25% d'hopitaux avec les salaires horaires les plus élevés.(>22.083750)

In [189]:
VAR_CTRL_2 = VAR_CTRL.copy()

VAR_CTRL_2 = VAR_CTRL_2.apply(lambda x: np.log(x) if x.name in ['Points_ISA', 'Capital_Lit_MCO', 'Capital_Bloc', 'Capital_imagerie',
       'Capital_Examens_biologiques', 'Capital_telemed', 'Population'] else x)

#On remplace les valeurs -infini apparues après l'application du log par la valeur -3 c'est-à-dire log(0.001)

from numpy import inf
VAR_CTRL_2.replace(-inf, inf, inplace = True)

for i in ['Points_ISA', 'Capital_Lit_MCO', 'Capital_Bloc', 'Capital_imagerie',
       'Capital_Examens_biologiques', 'Capital_telemed', 'Population']:
    VAR_CTRL_2[i]=VAR_CTRL_2[i].replace(inf, VAR_CTRL_2[i].min() - 0.01)


C:\Users\Vinciane\Anaconda3\lib\site-packages\pandas\core\arraylike.py:358: RuntimeWarning: divide by zero encountered in log
  result = getattr(ufunc, method)(*inputs, **kwargs)


In [190]:
# Les établissements avec des revenus départements  trés élevés

VAR_CTRL_ELEVES_2 = VAR_CTRL_2.copy()
VAR_CTRL_ELEVES_2= VAR_CTRL_ELEVES_2.rename(columns = {'Dep Moyenne Salaires Cadre Horaires': 'sal_cadre_moyen'})

VAR_CTRL_ELEVES_2= VAR_CTRL_ELEVES_2[VAR_CTRL_ELEVES_2.sal_cadre_moyen >= 21.280808]

In [191]:
# Les établissements avec des revenus départements trésfaibles

VAR_CTRL_FAIBLES_2 = VAR_CTRL_2.copy()
VAR_CTRL_FAIBLES_2 = VAR_CTRL_FAIBLES_2.rename(columns = {'Dep Moyenne Salaires Cadre Horaires': 'sal_cadre_moyen'})

VAR_CTRL_FAIBLES_2 = VAR_CTRL_FAIBLES_2[VAR_CTRL_FAIBLES_2.sal_cadre_moyen <= 21.280808]

In [192]:
VAR_CTRL_FAIBLES_2= VAR_CTRL_FAIBLES_2.fillna(0)
VAR_CTRL_ELEVES_2= VAR_CTRL_ELEVES_2.fillna(0)
VAR_EXPL_CAT_2 = VAR_EXPL_CAT_2.fillna(0)
#stjr_1 = stjr_1.fillna(0)

In [193]:
analyse_sal_faibles = pd.merge(VAR_CTRL_FAIBLES_2, VAR_EXPL_CAT_2)

analyse_sal_eleves = pd.merge(VAR_CTRL_ELEVES_2, VAR_EXPL_CAT_2)

In [194]:
analyse_sal_faibles_X =analyse_sal_faibles[["Personnel médical salarié",'Personnel médical libéral',"Internes",'Personnels de direction et administratifs',
       'Personnels des services de soins', 'Personnels éducatifs et sociaux',
       'Personnels médico-techniques', 'Personnels techniques et ouvriers', 
        'Capital_Lit_MCO', 'Capital_Bloc', 'Capital_imagerie', 'Population',
       'Capital_Examens_biologiques', 'Capital_telemed', 'STJR_1','STJR_2','STJR_3']]
analyse_sal_faibles_Y =analyse_sal_faibles[['Points_ISA']]
analyse_sal_eleves_X =analyse_sal_eleves[["Personnel médical salarié",'Personnel médical libéral',"Internes",'Personnels de direction et administratifs',
       'Personnels des services de soins', 'Personnels éducatifs et sociaux',
       'Personnels médico-techniques', 'Personnels techniques et ouvriers', 
       'Capital_Lit_MCO', 'Capital_Bloc', 'Capital_imagerie', 'Population',
       'Capital_Examens_biologiques', 'Capital_telemed', 'STJR_1','STJR_2','STJR_3']]
analyse_dep_eleves_Y =analyse_sal_eleves[['Points_ISA']]

In [195]:
analyse_sal_faibles_X = sm.add_constant(analyse_sal_faibles_X) # on ajoute une constante, sinon la régression se fait sans, par défaut
y = analyse_sal_faibles[['Points_ISA']]

model_sal_faibles = sm.OLS(y, analyse_sal_faibles_X)
results_sal_faibles = model_sal_faibles.fit()
print(results_sal_faibles.summary())

                            OLS Regression Results                            
Dep. Variable:             Points_ISA   R-squared:                       0.893
Model:                            OLS   Adj. R-squared:                  0.890
Method:                 Least Squares   F-statistic:                     279.2
Date:                Sun, 16 May 2021   Prob (F-statistic):          2.06e-247
Time:                        17:09:37   Log-Likelihood:                -487.48
No. Observations:                 552   AIC:                             1009.
Df Residuals:                     535   BIC:                             1082.
Df Model:                          16                                         
Covariance Type:            nonrobust                                         
                                                coef    std err          t      P>|t|      [0.025      0.975]
----------------------------------------------------------------------------------------------------

In [196]:
analyse_sal_eleves_X = sm.add_constant(analyse_sal_eleves_X) # on ajoute une constante, sinon la régression se fait sans, par défaut
y = analyse_sal_eleves[['Points_ISA']]

model_sal_eleves = sm.OLS(y, analyse_sal_eleves_X)
results_sal_eleves = model_sal_eleves.fit()
print(results_sal_eleves.summary())

                            OLS Regression Results                            
Dep. Variable:             Points_ISA   R-squared:                       0.888
Model:                            OLS   Adj. R-squared:                  0.885
Method:                 Least Squares   F-statistic:                     288.1
Date:                Sun, 16 May 2021   Prob (F-statistic):          2.47e-264
Time:                        17:09:38   Log-Likelihood:                -519.11
No. Observations:                 600   AIC:                             1072.
Df Residuals:                     583   BIC:                             1147.
Df Model:                          16                                         
Covariance Type:            nonrobust                                         
                                                coef    std err          t      P>|t|      [0.025      0.975]
----------------------------------------------------------------------------------------------------

# Troisième analyse : effet du salaire moyen département des employés sur la participation des personnels des services de soins et des services techniques

In [197]:
VAR_EXPL_CAT_3 = VAR_EXPL_CAT.copy()

In [198]:
VAR_EXPL_CAT_3['Personnel médical salarié'] = VAR_EXPL_CAT_3['Médecins salariés'] + VAR_EXPL_CAT_3["Autres personnels médicaux salariés"]
VAR_EXPL_CAT_3['Personnel médical libéral'] = VAR_EXPL_CAT_3['Médecins libéraux'] + VAR_EXPL_CAT_3["Autres personnels médicaux libéraux"]

In [199]:
VAR_EXPL_CAT_3 = VAR_EXPL_CAT_3[['Numero_finess', 'Personnel médical salarié','Personnel médical libéral','stjr',
       'Personnels de direction et administratifs',
       'Personnels des services de soins', 'Personnels éducatifs et sociaux',
       'Personnels médico-techniques', 'Personnels techniques et ouvriers',
        'Internes', 'STJR_1','STJR_2','STJR_3']]
VAR_EXPL_CAT_3= VAR_EXPL_CAT_3.fillna(0)

In [200]:
VAR_EXPL_CAT_3 = VAR_EXPL_CAT_3.apply(lambda x: np.log(x) if x.name in ['Personnel médical salarié','Personnel médical libéral',
       'Personnels de direction et administratifs',
       'Personnels des services de soins', 'Personnels éducatifs et sociaux',
       'Personnels médico-techniques', 'Personnels techniques et ouvriers',
        'Internes'] else x)

#On remplace les valeurs -infini apparues après l'application du log par la valeur -3 c'est-à-dire log(0.001)
from numpy import inf
VAR_EXPL_CAT_3.replace(-inf, inf, inplace = True)

for i in ['Personnels de direction et administratifs','Personnels des services de soins', 'Personnels éducatifs et sociaux',
                                                    'Personnels médico-techniques', 'Personnels techniques et ouvriers',
                                                    'Personnel médical salarié','Personnel médical libéral', 'Internes']:
    VAR_EXPL_CAT_3[i]=VAR_EXPL_CAT_3[i].replace(inf, VAR_EXPL_CAT_3[i].min() - 0.01)


C:\Users\Vinciane\Anaconda3\lib\site-packages\pandas\core\arraylike.py:358: RuntimeWarning: divide by zero encountered in log
  result = getattr(ufunc, method)(*inputs, **kwargs)


In [201]:
VAR_CTRL.describe()

Points_ISA         stjr  Capital_Lit_MCO  Capital_Bloc  \
count  1.162000e+03  1162.000000      1139.000000   1153.000000   
mean   2.939915e+05     1.707401       125.109263      7.379577   
std    6.281782e+05     0.716795       127.684194      7.191520   
min    1.347900e+02     1.000000         0.000000      0.000000   
25%    2.993899e+04     1.000000        35.000000      1.000000   
50%    1.016205e+05     2.000000        85.000000      6.000000   
75%    2.975944e+05     2.000000       171.166667     10.500000   
max    9.116903e+06     3.000000      1055.000000     44.000000   

       Capital_imagerie  Capital_Examens_biologiques  Capital_telemed  \
count       1162.000000                 1.149000e+03      1162.000000   
mean           4.256971                 2.508881e+05         0.728148   
std            5.047103                 6.567670e+05         1.514649   
min            0.000000                 0.000000e+00         0.000000   
25%            0.000000                 0.000000e+00         0.000000   
50%            3.000000                 0.000000e+00         0.000000   
75%            7.000000                 6.261400e+04         1.000000   
max           28.000000                 1.060114e+07        11.000000   

          SEJHC_MED     SEJHC_CHI    SEJHC_OBS     SEJHC_MCO  \
count   1162.000000   1162.000000  1161.000000   1162.000000   
mean    3953.236728   2091.528256   673.093714   6717.279443   
std     5172.505167   2471.292213  1146.876402   7600.700172   
min        0.000000      0.000000     0.000000      0.000000   
25%      364.750000      0.000000     0.000000   1204.500000   
50%     1963.000000   1428.000000     0.000000   4168.500000   
75%     5581.250000   3168.312500  1081.000000   9749.750000   
max    39884.000000  20991.000000  7639.000000  60719.000000   

       Taux_chomage_par_dep    Population  Dep Moyenne Salaires Horaires  \
count           1162.000000  1.162000e+03                    1162.000000   
mean               8.290103  1.367009e+05                      11.793749   
std                1.660836  3.998901e+05                       1.908702   
min                5.000000  1.820000e+02                       9.862500   
25%                7.000000  8.053750e+03                      10.633333   
50%                8.000000  2.295500e+04                      11.204762   
75%                9.300000  6.171825e+04                      11.981429   
max               13.800000  2.125851e+06                      17.800000   

       Dep Moyenne Salaires Cadre Horaires  \
count                          1162.000000   
mean                             21.764363   
std                               1.674054   
min                              19.750000   
25%                              20.775000   
50%                              21.280808   
75%                              22.083750   
max                              27.600000   

       Dep Moyenne Salaires Prof Intermédiaire Horaires  \
count                                       1162.000000   
mean                                          12.674821   
std                                            0.523229   
min                                           11.740000   
25%                                           12.346721   
50%                                           12.535714   
75%                                           12.902500   
max                                           14.185000   

       Dep Moyenne Salaires Employé Horaires  \
count                            1162.000000   
mean                                8.783985   
std                                 0.353571   
min                                 8.261905   
25%                                 8.555172   
50%                                 8.700862   
75%                                 8.848113   
max                                 9.742553   

       Moyenne Revenus Fiscaux Départementaux  
count                             1162.000000  
mean     

In [202]:
VAR_CTRL_3 = VAR_CTRL.copy()

VAR_CTRL_3 = VAR_CTRL_3.apply(lambda x: np.log(x) if x.name in ['Points_ISA', 'Capital_Lit_MCO', 'Capital_Bloc', 'Capital_imagerie',
       'Capital_Examens_biologiques', 'Capital_telemed', 'Population'] else x)

#On remplace les valeurs -infini apparues après l'application du log par la valeur -3 c'est-à-dire log(0.001)
from numpy import inf
VAR_CTRL_3.replace(-inf, inf, inplace = True)

for i in ['Points_ISA', 'Capital_Lit_MCO', 'Capital_Bloc', 'Capital_imagerie',
       'Capital_Examens_biologiques', 'Capital_telemed', 'Population']:
    VAR_CTRL_3[i]=VAR_CTRL_3[i].replace(inf, VAR_CTRL_3[i].min() - 0.01)





C:\Users\Vinciane\Anaconda3\lib\site-packages\pandas\core\arraylike.py:358: RuntimeWarning: divide by zero encountered in log
  result = getattr(ufunc, method)(*inputs, **kwargs)


On garde les 25% d'hopitaux avec les salaires horaires des employés les plus bas (<8.555172, et les 25% d'hopitaux avec les salaires horaires les plus élevés.(>8.848113)

In [203]:
# Les établissements avec des revenus départements  trés élevés

VAR_CTRL_ELEVES_3 = VAR_CTRL_3.copy()
VAR_CTRL_ELEVES_3= VAR_CTRL_ELEVES_3.rename(columns = {'Dep Moyenne Salaires Employé Horaires': 'sal_employes_moyen'})

VAR_CTRL_ELEVES_3= VAR_CTRL_ELEVES_3[VAR_CTRL_ELEVES_3.sal_employes_moyen > 8.700862]

In [204]:
# Les établissements avec des revenus départements trésfaibles

VAR_CTRL_FAIBLES_3 = VAR_CTRL_3.copy()
VAR_CTRL_FAIBLES_3 = VAR_CTRL_FAIBLES_3.rename(columns = {'Dep Moyenne Salaires Employé Horaires': 'sal_employe_moyen'})

VAR_CTRL_FAIBLES_3 = VAR_CTRL_FAIBLES_3[VAR_CTRL_FAIBLES_3.sal_employe_moyen <= 8.700862]

In [205]:
VAR_CTRL_FAIBLES_3= VAR_CTRL_FAIBLES_3.fillna(0)
VAR_CTRL_ELEVES_3= VAR_CTRL_ELEVES_3.fillna(0)
VAR_EXPL_CAT_3 = VAR_EXPL_CAT_3.fillna(0)
#stjr_1 = stjr_1.fillna(0)

In [206]:
analyse_sal_faibles_3 = pd.merge(VAR_CTRL_FAIBLES_3, VAR_EXPL_CAT_3)

analyse_sal_eleves_3 = pd.merge(VAR_CTRL_ELEVES_3, VAR_EXPL_CAT_3)

In [207]:
analyse_sal_faibles_X_3 =analyse_sal_faibles_3[["Personnel médical salarié",'Personnel médical libéral',"Internes",'Personnels de direction et administratifs',
       'Personnels des services de soins', 'Personnels éducatifs et sociaux',
       'Personnels médico-techniques', 'Personnels techniques et ouvriers', 
        'Capital_Lit_MCO', 'Capital_Bloc', 'Capital_imagerie', 'Population',
       'Capital_Examens_biologiques', 'Capital_telemed', 'STJR_1','STJR_2','STJR_3']]
analyse_sal_faibles_Y_3 =analyse_sal_faibles_3[['Points_ISA']]
analyse_sal_eleves_X_3 =analyse_sal_eleves_3[["Personnel médical salarié",'Personnel médical libéral',"Internes",'Personnels de direction et administratifs',
       'Personnels des services de soins', 'Personnels éducatifs et sociaux',
       'Personnels médico-techniques', 'Personnels techniques et ouvriers', 
       'Capital_Lit_MCO', 'Capital_Bloc', 'Capital_imagerie', 'Population',
       'Capital_Examens_biologiques', 'Capital_telemed', 'STJR_1','STJR_2','STJR_3']]
analyse_dep_eleves_Y_3=analyse_sal_eleves_3[['Points_ISA']]

In [208]:
analyse_sal_faibles_X_3

Personnel médical salarié  Personnel médical libéral  Internes  \
0                     1.979621                   3.941582 -1.437116   
1                    -3.228876                   4.119037 -1.437116   
2                     3.813969                   3.511545 -1.437116   
3                    -3.228876                  -0.703147 -1.437116   
4                     5.857247                  -0.703147  5.540969   
..                         ...                        ...       ...   
560                   3.231989                  -0.703147  0.667829   
561                   3.932120                   2.014903  2.890372   
562                  -0.105361                   1.098612 -1.437116   
563                  -1.386294                   1.871802 -1.437116   
564                   0.009950                   1.098612 -1.437116   

     Personnels de direction et administratifs  \
0                                     4.932169   
1                                     3.933197   
2                                     4.466598   
3                                     2.446685   
4                                     5.296841   
..                                         ...   
560                                   3.411148   
561                                   3.539993   
562                                   1.945910   
563                                   0.565314   
564                                   1.249902   

     Personnels des services de soins  Personnels éducatifs et sociaux  \
0                            5.898554                        -3.922023   
1                            5.933412                        -0.223144   
2                            5.874144                         0.350657   
3                            4.843793                         1.004302   
4                            7.050556                         3.021644   
..                                ...                              ...   
560                          5.165072                         0.693147   
561                          4.997888                         1.483875   
562                          3.837299                         0.000000   
563                          3.149311                        -3.922023   
564                          3.651697                        -1.714798   

     Personnels médico-techniques  Personnels techniques et ouvriers  \
0                        4.252630                           2.702032   
1                        2.304583                           1.420696   
2                        3.488903                           2.917230   
3                        0.000000                           1.321756   
4                        4.712544                           5.358259   
..                            ...                                ...   
560                      3.397858                           3.632309   
561                      2.924236                           3.096256   
562                      0.000000                           2.420368   
563                     -1.386294                           1.324419   
564                     -3.228876                           1.916923   

     Capital_Lit_MCO  Capital_Bloc  Capital_imagerie  Population  \
0           5.111988      2.772589          2.397895   11.609517   
1           5.313206      2.564949         -0.010000   11.609517   
2           5.267858      2.564949          2.302585   11.083956   
3           2.890372     -0.010000          0.000000   11.083956   
4           5.993961      3.784190          3.135494   11.083956   
..               ...           ...               ...         ...   
560         3.526361      1.609438          1.098612    8.822764   
561         4.204693      1.504077          1.791759    9.375177   
562         3.135494     -0.010000          0.000000    8.038189   
563         3.091042     -0.010000         -0.010000    8.370779   
564         2.995732     -0.010000         -0.010

In [209]:
analyse_sal_faibles_X_3 = sm.add_constant(analyse_sal_faibles_X_3) # on ajoute une constante, sinon la régression se fait sans, par défaut
y = analyse_sal_faibles_3[['Points_ISA']]

model_sal_faibles_3 = sm.OLS(y, analyse_sal_faibles_X_3)
results_sal_faibles_3 = model_sal_faibles.fit()
print(results_sal_faibles_3.summary())

                            OLS Regression Results                            
Dep. Variable:             Points_ISA   R-squared:                       0.893
Model:                            OLS   Adj. R-squared:                  0.890
Method:                 Least Squares   F-statistic:                     279.2
Date:                Sun, 16 May 2021   Prob (F-statistic):          2.06e-247
Time:                        17:11:01   Log-Likelihood:                -487.48
No. Observations:                 552   AIC:                             1009.
Df Residuals:                     535   BIC:                             1082.
Df Model:                          16                                         
Covariance Type:            nonrobust                                         
                                                coef    std err          t      P>|t|      [0.025      0.975]
----------------------------------------------------------------------------------------------------

In [210]:
analyse_sal_eleves_X_3 = sm.add_constant(analyse_sal_eleves_X_3) # on ajoute une constante, sinon la régression se fait sans, par défaut
y = analyse_sal_eleves_3[['Points_ISA']]

model_sal_eleves_3 = sm.OLS(y, analyse_sal_eleves_X_3)
results_sal_eleves_3 = model_sal_eleves.fit()
print(results_sal_eleves_3.summary())

                            OLS Regression Results                            
Dep. Variable:             Points_ISA   R-squared:                       0.888
Model:                            OLS   Adj. R-squared:                  0.885
Method:                 Least Squares   F-statistic:                     288.1
Date:                Sun, 16 May 2021   Prob (F-statistic):          2.47e-264
Time:                        17:11:02   Log-Likelihood:                -519.11
No. Observations:                 600   AIC:                             1072.
Df Residuals:                     583   BIC:                             1147.
Df Model:                          16                                         
Covariance Type:            nonrobust                                         
                                                coef    std err          t      P>|t|      [0.025      0.975]
----------------------------------------------------------------------------------------------------

# Taille des villes : est ce la participation aux Points-ISA est la même dans les grandes et petites villes?

On prend les 25% des villes avec le moins d'ahbitants, par rapport aux 25% des villes avec le plus d'habitants

In [211]:
VAR_EXPL_CAT_4 = VAR_EXPL_CAT.copy()

In [212]:
VAR_EXPL_CAT_4 = VAR_EXPL_CAT_4[['Numero_finess', 'Personnel médical','stjr',
       'Personnels de direction et administratifs',
       'Personnels des services de soins', 'Personnels éducatifs et sociaux',
       'Personnels médico-techniques', 'Personnels techniques et ouvriers',
        'Internes', 'STJR_1','STJR_2','STJR_3']]
VAR_EXPL_CAT_4= VAR_EXPL_CAT_4.fillna(0)

In [213]:
VAR_EXPL_CAT_4 = VAR_EXPL_CAT_4.apply(lambda x: np.log(x) if x.name in ['Personnel médical',
       'Personnels de direction et administratifs',
       'Personnels des services de soins', 'Personnels éducatifs et sociaux',
       'Personnels médico-techniques', 'Personnels techniques et ouvriers',
        'Internes'] else x)

from numpy import inf
VAR_EXPL_CAT_4.replace(-inf, inf, inplace = True)

for i in ['Personnels de direction et administratifs','Personnels des services de soins', 'Personnels éducatifs et sociaux',
                                                    'Personnels médico-techniques', 'Personnels techniques et ouvriers',
                                                    'Personnel médical', 'Internes']:
    VAR_EXPL_CAT_4[i]=VAR_EXPL_CAT_4[i].replace(inf, VAR_EXPL_CAT_4[i].min() - 0.01)


C:\Users\Vinciane\Anaconda3\lib\site-packages\pandas\core\arraylike.py:358: RuntimeWarning: divide by zero encountered in log
  result = getattr(ufunc, method)(*inputs, **kwargs)


In [214]:
VAR_CTRL.describe()

Points_ISA         stjr  Capital_Lit_MCO  Capital_Bloc  \
count  1.162000e+03  1162.000000      1139.000000   1153.000000   
mean   2.939915e+05     1.707401       125.109263      7.379577   
std    6.281782e+05     0.716795       127.684194      7.191520   
min    1.347900e+02     1.000000         0.000000      0.000000   
25%    2.993899e+04     1.000000        35.000000      1.000000   
50%    1.016205e+05     2.000000        85.000000      6.000000   
75%    2.975944e+05     2.000000       171.166667     10.500000   
max    9.116903e+06     3.000000      1055.000000     44.000000   

       Capital_imagerie  Capital_Examens_biologiques  Capital_telemed  \
count       1162.000000                 1.149000e+03      1162.000000   
mean           4.256971                 2.508881e+05         0.728148   
std            5.047103                 6.567670e+05         1.514649   
min            0.000000                 0.000000e+00         0.000000   
25%            0.000000                 0.000000e+00         0.000000   
50%            3.000000                 0.000000e+00         0.000000   
75%            7.000000                 6.261400e+04         1.000000   
max           28.000000                 1.060114e+07        11.000000   

          SEJHC_MED     SEJHC_CHI    SEJHC_OBS     SEJHC_MCO  \
count   1162.000000   1162.000000  1161.000000   1162.000000   
mean    3953.236728   2091.528256   673.093714   6717.279443   
std     5172.505167   2471.292213  1146.876402   7600.700172   
min        0.000000      0.000000     0.000000      0.000000   
25%      364.750000      0.000000     0.000000   1204.500000   
50%     1963.000000   1428.000000     0.000000   4168.500000   
75%     5581.250000   3168.312500  1081.000000   9749.750000   
max    39884.000000  20991.000000  7639.000000  60719.000000   

       Taux_chomage_par_dep    Population  Dep Moyenne Salaires Horaires  \
count           1162.000000  1.162000e+03                    1162.000000   
mean               8.290103  1.367009e+05                      11.793749   
std                1.660836  3.998901e+05                       1.908702   
min                5.000000  1.820000e+02                       9.862500   
25%                7.000000  8.053750e+03                      10.633333   
50%                8.000000  2.295500e+04                      11.204762   
75%                9.300000  6.171825e+04                      11.981429   
max               13.800000  2.125851e+06                      17.800000   

       Dep Moyenne Salaires Cadre Horaires  \
count                          1162.000000   
mean                             21.764363   
std                               1.674054   
min                              19.750000   
25%                              20.775000   
50%                              21.280808   
75%                              22.083750   
max                              27.600000   

       Dep Moyenne Salaires Prof Intermédiaire Horaires  \
count                                       1162.000000   
mean                                          12.674821   
std                                            0.523229   
min                                           11.740000   
25%                                           12.346721   
50%                                           12.535714   
75%                                           12.902500   
max                                           14.185000   

       Dep Moyenne Salaires Employé Horaires  \
count                            1162.000000   
mean                                8.783985   
std                                 0.353571   
min                                 8.261905   
25%                                 8.555172   
50%                                 8.700862   
75%                                 8.848113   
max                                 9.742553   

       Moyenne Revenus Fiscaux Départementaux  
count                             1162.000000  
mean     

Les 25% des villes avec le moins d'habitants ont une population inférieure 8.055250e+03, tandis que les 25% des vilels avec le plus d'habitants, ont plus de 6.227500e+04 habitants

In [215]:
VAR_CTRL_4 = VAR_CTRL.copy()

In [216]:
# Les établissements avec des revenus départements  trés élevés

VAR_CTRL_ELEVES_4 = VAR_CTRL_4.copy()

VAR_CTRL_ELEVES_4= VAR_CTRL_ELEVES_4[VAR_CTRL_ELEVES_4.Population >= 2.295500e+04]

In [217]:

VAR_CTRL_ELEVES_4 = VAR_CTRL_ELEVES_4.apply(lambda x: np.log(x) if x.name in ['Points_ISA', 'Capital_Lit_MCO', 'Capital_Bloc', 'Capital_imagerie',
       'Capital_Examens_biologiques', 'Capital_telemed', 'Population'] else x)

#On remplace les valeurs -infini apparues après l'application du log par la valeur -3 c'est-à-dire log(0.001)
from numpy import inf
VAR_CTRL_ELEVES_4.replace(-inf, inf, inplace = True)

for i in ['Points_ISA', 'Capital_Lit_MCO', 'Capital_Bloc', 'Capital_imagerie',
       'Capital_Examens_biologiques', 'Capital_telemed', 'Population']:
    VAR_CTRL_ELEVES_4[i]=VAR_CTRL_ELEVES_4[i].replace(inf, VAR_CTRL_ELEVES_4[i].min() - 0.01)




C:\Users\Vinciane\Anaconda3\lib\site-packages\pandas\core\arraylike.py:358: RuntimeWarning: divide by zero encountered in log
  result = getattr(ufunc, method)(*inputs, **kwargs)


In [218]:
# Les établissements avec des revenus départements trésfaibles

VAR_CTRL_FAIBLES_4 = VAR_CTRL_4.copy()
VAR_CTRL_FAIBLES_4 = VAR_CTRL_FAIBLES_4[VAR_CTRL_FAIBLES_4.Population <= 2.295500e+04]

In [219]:
VAR_CTRL_FAIBLES_4 = VAR_CTRL_FAIBLES_4.apply(lambda x: np.log(x) if x.name in ['Points_ISA', 'Capital_Lit_MCO', 'Capital_Bloc', 'Capital_imagerie',
       'Capital_Examens_biologiques', 'Capital_telemed', 'Population'] else x)

#On remplace les valeurs -infini apparues après l'application du log par la valeur -3 c'est-à-dire log(0.001)
from numpy import inf
VAR_CTRL_FAIBLES_4.replace(-inf, inf, inplace = True)

for i in ['Points_ISA', 'Capital_Lit_MCO', 'Capital_Bloc', 'Capital_imagerie',
       'Capital_Examens_biologiques', 'Capital_telemed', 'Population']:
    VAR_CTRL_FAIBLES_4[i]=VAR_CTRL_FAIBLES_4[i].replace(inf, VAR_CTRL_FAIBLES_4[i].min() - 0.01)



C:\Users\Vinciane\Anaconda3\lib\site-packages\pandas\core\arraylike.py:358: RuntimeWarning: divide by zero encountered in log
  result = getattr(ufunc, method)(*inputs, **kwargs)


In [220]:
VAR_CTRL_FAIBLES_4= VAR_CTRL_FAIBLES_4.fillna(0)
VAR_CTRL_ELEVES_4= VAR_CTRL_ELEVES_4.fillna(0)
VAR_EXPL_CAT_4 = VAR_EXPL_CAT_4.fillna(0)
#stjr_1 = stjr_1.fillna(0)

In [221]:
analyse_pop_faibles_4 = pd.merge(VAR_CTRL_FAIBLES_4, VAR_EXPL_CAT_4)

analyse_pop_eleves_4 = pd.merge(VAR_CTRL_ELEVES_4, VAR_EXPL_CAT_4)

In [222]:
analyse_pop_eleves_4 

Numero_finess  Points_ISA  stjr dep            Departement COMINSEE  \
0       060000528   12.785189     3  06        Alpes-Maritimes    06088   
1       060780947   12.069981     3  06        Alpes-Maritimes    06088   
2       060791811   11.894321     3  06        Alpes-Maritimes    06088   
3       060780442   10.036373     2  06        Alpes-Maritimes    06088   
4       060780715   12.742716     2  06        Alpes-Maritimes    06088   
..            ...         ...   ...  ..                    ...      ...   
569     610780082   13.140029     1  61                   Orne    61001   
570     850000118   12.089706     2  85                 Vendée    85191   
571     850000019   14.232458     1  85                 Vendée    85191   
572     900000035   11.126485     2  90  Territoire de Belfort    90010   
573     120780044   13.518520     1  12                Aveyron    12202   

     Capital_Lit_MCO  Capital_Bloc  Capital_imagerie  \
0           4.852030      2.079442          2.197225   
1           4.653960      1.609438          1.386294   
2           4.510860      0.693147          0.693147   
3           3.401197      1.791759          0.000000   
4           5.545177      3.258097          2.302585   
..               ...           ...               ...   
569         5.000585      1.945910          2.140066   
570         4.700480      2.397895         -0.010000   
571         5.542570      3.091042          2.427748   
572         3.850148      2.197225         -0.010000   
573         5.733341      2.639057          2.302585   

     Capital_Examens_biologiques  ...  Personnel médical  \
0                       6.812345  ...           4.512287   
1                       5.707028  ...           4.289089   
2                       5.707028  ...           3.331133   
3                       5.707028  ...           3.465736   
4                       5.707028  ...           4.844187   
..                           ...  ...                ...   
569                    13.689513  ...           4.639700   
570                     5.707028  ...           3.618993   
571                    13.712303  ...           5.036411   
572                     5.707028  ...           3.054001   
573                    13.767212  ...           4.880957   

     Personnels de direction et administratifs  \
0                                     4.983333   
1                                     4.721708   
2                                     3.820565   
3                                     2.833213   
4                                     4.423768   
..                                         ...   
569                                   4.017524   
570                                   3.281287   
571                                   4.892152   
572                                   3.030134   
573                                   4.132282   

     Personnels des services de soins  Personnels éducatifs et sociaux  \
0                            5.774707                         1.510722   
1                            5.860444                         0.810930   
2                            5.505941                         1.278152   
3                            4.320151                         0.000000   
4                            6.041207                        -0.301105   
..                                ...                              ...   
569                          5.780723                         1.040277   
570                          5.189228                        -0.223144   
571                          6.667622                         2.142808   
572                          4.750482                         0.000000   
573                          5.688161                         2.319114   

     Personnels médico-techniques  Personnels techniques et ouvriers  \
0                        5.106794                           3.672242   
1                        3.074543                           2.955431   
2  

In [223]:
analyse_pop_faibles_X_4 =analyse_pop_faibles_4[["Personnel médical","Internes",'Personnels de direction et administratifs',
       'Personnels des services de soins', 'Personnels éducatifs et sociaux',
       'Personnels médico-techniques', 'Personnels techniques et ouvriers', 
        'Capital_Lit_MCO', 'Capital_Bloc', 'Capital_imagerie',
       'Capital_Examens_biologiques', 'Capital_telemed', 'STJR_1','STJR_2','STJR_3']]
analyse_pop_faibles_Y_4 =analyse_pop_faibles_4[['Points_ISA']]
analyse_pop_eleves_X_4 =analyse_pop_eleves_4[["Personnel médical","Internes",'Personnels de direction et administratifs',
       'Personnels des services de soins', 'Personnels éducatifs et sociaux',
       'Personnels médico-techniques', 'Personnels techniques et ouvriers', 
       'Capital_Lit_MCO', 'Capital_Bloc', 'Capital_imagerie',
       'Capital_Examens_biologiques', 'Capital_telemed', 'STJR_1','STJR_2','STJR_3']]
analyse_pop_eleves_Y_4=analyse_pop_eleves_4[['Points_ISA']]

In [224]:
analyse_pop_faibles_X_4 = sm.add_constant(analyse_pop_faibles_X_4) # on ajoute une constante, sinon la régression se fait sans, par défaut
y = analyse_pop_faibles_4[['Points_ISA']]

model_pop_faibles_4 = sm.OLS(y, analyse_pop_faibles_X_4)
results_pop_faibles_4 = model_pop_faibles_4.fit()
print(results_pop_faibles_4.summary())

                            OLS Regression Results                            
Dep. Variable:             Points_ISA   R-squared:                       0.878
Model:                            OLS   Adj. R-squared:                  0.875
Method:                 Least Squares   F-statistic:                     289.4
Date:                Sun, 16 May 2021   Prob (F-statistic):          2.25e-246
Time:                        17:11:51   Log-Likelihood:                -527.35
No. Observations:                 578   AIC:                             1085.
Df Residuals:                     563   BIC:                             1150.
Df Model:                          14                                         
Covariance Type:            nonrobust                                         
                                                coef    std err          t      P>|t|      [0.025      0.975]
----------------------------------------------------------------------------------------------------

In [225]:
analyse_pop_eleves_X_4 = sm.add_constant(analyse_pop_eleves_X_4) # on ajoute une constante, sinon la régression se fait sans, par défaut
y = analyse_pop_eleves_4[['Points_ISA']]

model_pop_eleves_4 = sm.OLS(y, analyse_pop_eleves_X_4)
results_pop_eleves_4 = model_pop_eleves_4.fit()
print(results_pop_eleves_4.summary())

                            OLS Regression Results                            
Dep. Variable:             Points_ISA   R-squared:                       0.901
Model:                            OLS   Adj. R-squared:                  0.899
Method:                 Least Squares   F-statistic:                     363.6
Date:                Sun, 16 May 2021   Prob (F-statistic):          6.35e-270
Time:                        17:11:51   Log-Likelihood:                -412.82
No. Observations:                 574   AIC:                             855.6
Df Residuals:                     559   BIC:                             920.9
Df Model:                          14                                         
Covariance Type:            nonrobust                                         
                                                coef    std err          t      P>|t|      [0.025      0.975]
----------------------------------------------------------------------------------------------------